Boltdir/site-modules/metrics/files/collect_metrics_files.rb

```diff
@@ -0,0 +1,370 @@
+    # @param [string] service name of the service the directory belongs to
+    #
+    # @return [void]
+    #
+    # @example
+    #   inspect_service_dir_for_metrics_files(dir_path, service)
+    def inspect_service_dir_for_metrics_files(dir_path, service)
+      service_dir = Dir.open dir_path
+      service_dir.each do |path|
+        next if path =~ /^\.$|^\.\.$/
+
+        absolute_path = "#{service_dir.path}/#{path}"
+        if File.directory?(absolute_path)
+          inspect_service_dir_for_metrics_files(absolute_path, service)
+          next
+        end
+        if path =~ /.json$/
+          consider_json_for_inclusion(absolute_path)
+        elsif path =~ /.tar.gz$/
+          consider_tarfile_contents_for_inclusion(absolute_path, service)
+        end
+      end
+    end
+
+    # Determines if a json file is from the target window
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] file_path json file_path
+    #
+    # @return [boolean] true if yes, false if no
+    #
+    # @example
+    #   json_file_from_target_window(file_path)
+    def json_file_from_target_window?(file_path)
+      raise ArgumentError, "Argument is not a json filename: #{file_path}" unless file_path.match(/\.json$/i)
+
+      time_str = File.basename(file_path, ".json")
+      time_obj = DateTime.parse(time_str)
+      epoch = time_obj.to_time.to_i
+      return epoch > @start_epoch && (epoch - @poll_interval) < @end_epoch
+    end
+
+    # Determines if a json file is from the target window and if so, stages it to the staging_dir
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] file_path absolute path to json file
+    #
+    # @return [void]
+    #
+    # @example
+    #   consider_json_for_inclusion(file_path)
+    def consider_json_for_inclusion(file_path)
+      stage_json(file_path) if json_file_from_target_window?(file_path)
+    end
+
+    # Stage json to the output_dir
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] file_path absolute path to json file
+    #
+    # @return [void]
+    #
+    # @example
+    #   stage_json(file_path)
+    def stage_json(file_path)
+      file_dir, filename = File.split(file_path)
+      relative_dir_path = file_dir.gsub(%r{^#{@metrics_dir}/}, "")
+      destination_dir = "#{@staging_dir}/#{relative_dir_path}"
+      FileUtils.mkdir(destination_dir) unless File.directory?(destination_dir)
+      FileUtils.cp(file_path, "#{destination_dir}/#{filename}")
+    end
+
+    # Determines if a tar file contains jsons from the target window and if so,
+    # either copies the whole tar file, or extracts the jsons from it to copy the relevant ones to
+    # the output_dir
+    #
+    # @author Randell Pelak
+    #
+    # @param [string] file_path absolute path to tar file
+    # @param [string] service name of the service the file comes from
+    #
+    # @return [void]
+    #
+    # @example
+    #   process_tar_file(file_path, service)
```

Comment:
wrong method call in `@example`

Do we need these `@example`'s if they are only duplicating the method signature itself?  I don't see what value they add.